File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
'''
Defines a rectnagle
'''


class Rectangle:
    '''
    Define a rectangle and set the properties
    '''

    def __init__(self, width=0, height=0):
        self.__width = width
        self.__height = height

    @property
    def width(self):
        return self.__width

    @property
    def height(self):
        return self.__height

    @width.setter
    def width(self, value):
        if type(value) != int:
            raise TypeError('width must be an integer')
        if value < 0:
            raise ValueError('width must be >= 0')
        self.__width = value

    @height.setter
    def height(self, value):
        if type(value) != int:
            raise TypeError('height must be an integer')
        if value < 0:
            raise ValueError('height must be >= 0')
        self.__height = value

    def area(self):
        ''' returns the area of the rectagle '''

        return self.__height * self.__width

    def perimeter(self):
        ''' returns the erimeter of the rectangle '''

        if self.__height == 0 or self.__width == 0:
            return 0
        return 2 * (self.__height + self.__width)

    def __str__(self):
        string = ''
        if self.__width != 0 and self.__height != 0:
            string += '\n'.join('#' * self.__width
                                for i in range(self.__height))
        return string

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    if (w <= 0 || h <= 0 || w === undefined || h === undefined) {
      return this;
    } else {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    let rowStr;
    for (let i = 0; i < this.height; i++) {
      rowStr = '';
      for (let j = 0; j < this.width; j++) {
        rowStr += 'X';
      }
      console.log(rowStr);
    }
  }
}
module.exports = Rectangle;
